src/sentry/auth/helper.py

```diff
@@ -81,6 +81,12 @@ def mark_session(self):
         super().mark_session()
         self.request.session.modified = True
 
+    def is_valid(self):
+        return super().is_valid() and self.flow in (
+            AuthHelper.FLOW_LOGIN,
+            AuthHelper.FLOW_SETUP_PROVIDER,
+        )
```

Comment:
I've extended `is_valid()` here so that it'll be called here when we're unpacking state from redis: https://github.com/getsentry/sentry/blob/a0ec5a7af5835413781c5c26fa81fbe41b1e9879/src/sentry/pipeline/base.py#L73-L74